components/webrtc_view_bulma/RemoteVideoContainer.jsx

```diff
@@ -1,71 +1,72 @@
+// Copyright (c) 2018-present Riff Learning, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
 import React from 'react';
-import * as sc from './styled';
-import SharedScreen from './SharedScreen'
-import PeerVideo from './PeerVideo'
-import { logger } from '../../utils/riff';
 
+import {logger} from '../../utils/riff';
+
+import SharedScreen from './SharedScreen';
+import PeerVideo from './PeerVideo';
 
 class RemoteVideoContainer extends React.Component {
     constructor(props) {
         super(props);
-        console.log("remote video props:", props);
+        logger.debug('remote video props:', props);
     }
 
     peerVideo(peerLength) {
         // returns a function
         // close over peerLength
-        return function (peer) {
-          let [riffId, displayName] = peer.nick.split("|");
-          let riffIds = this.props.chat.webRtcRiffIds.sort();
-          logger.debug("riff ids:", riffIds);
-          const idx = riffIds.indexOf(riffId);
-          let peerColor = this.props.chat.peerColors[idx];
-          logger.debug("!!PEER COLOR:", peerColor, "IDX:", idx, "Riff ID:", riffId);
-          return (
-            <PeerVideo
-              key={peer.id}
-              id={peer.id}
-              videoEl={peer.videoEl}
-              type="peer"
-              peerColor={peerColor}
-              peerLength={peerLength}
-            />
-          );
-        }.bind(this)
+        return (peer) => {
+            const [riffId, displayName] = peer.nick.split('|'); // eslint-disable-line no-unused-vars
+            const riffIds = [...this.props.chat.webRtcRiffIds].sort();
```

Comment:
@iamtomhanks I see what you did here (creating a new array), what I'm not clear on is why. I'm sure you had a reason, I'd just like to understand it.

Reply:
Alright will do! I was trying to load 4 users at once and was sometimes getting an error relating to mutating the variable, and this solved it. I can't quite remember the error to be honest, but I think this is good practice, to not mutate the original array, if you're not intending to do so.

Reply:
Overall, yes, you should think hard about mutating the original array, and that's a good reason for your change, the fact that there was actually an error is worth noting.

Try to make a note of the error, and stick it in the commit message or something. Definitely not a must do, but I've found that occasionally it's been invaluable, when something similar arises and I can find it listed in the history, along with the fix. Or sometime in the future someone else decides there's no reason for the change and changes it back, and searching the repo history explains why.

Once you think my changes are good, this will be ready to be merged, but that probably won't happen until Tuesday, as I'm traveling on Monday. Just a heads up.